### PR TITLE
Timeseries testing update

### PR DESCRIPTION
**Pull request by wanghp18**
Update log:
- Fixed ngToast not shown bug.
- Add django-celery for `python manage.py celery` command to work.
- Move seed/energy/tasks.py to seed/energy/meter_data_processor folder to avoid circular import.
- Separate the code requesting GreenButton XML data as an independent task. Since schedulers will be set to run daily/monthly, when new GreenButton batch request info is saved by user, the new separated task will be called to request GreenButton XML file immediately instead of waiting for the scheduler to do it.
- Ignore rows missing critical data in uploaded energy template to avoid KairosDB insert exception.

Note:
- The backfill start date for GreenButton data is set to March 1, 2014, it is hard coded for the service provided by DC partner. I'm working on making it flexible.
- Processing Full PM exported spreadsheet data and energy template currently uses address_line_1 column to match building record in SEED, so the building records having PM or energy template data need to have the according address in address_line_1 column for now. I'm working on including building match module.
- Please don't modify the sheets and column structure of exported PM spreadsheet, the code parsing the report assuming the default structure.

### CODE REVIEW

File: config/settings/common.py [hunk 1]
```diff
@@ -222,15 +224,16 @@
 CELERY_RESULT_SERIALIZER = 'seed_json'
 CELERY_TASK_RESULT_EXPIRES = 18000  # 5 hours
 
+CELERY_IMPORTS = ('seed.energy.meter_data_processor.tasks')
 CELERYBEAT_SCHEDULE = {
-    'Run daily': {
-        'task': 'seed.energy.tasks.aggregate_monthly_data',
-        'schedule': timedelta(seconds=300), # For Demo purpose, should be days=1 in product environment
+    'Run monthly': {
+        'task': 'seed.energy.meter_data_processor.tasks.aggregate_monthly_data',
```
Comment (nllong) [inline config/settings/common.py]: I think we have moved this file a couple times. I think the tasks.py file for the energy 'app' is in /seed/energy/tasks.py

File: requirements/base.txt [hunk 2]
```diff
@@ -36,7 +36,7 @@ django-autoslug==1.8.0
 pytz==2012h
 
 # background process management
-# django-celery==3.1.16
+django-celery==3.1.16
```
Comment (nllong) [inline requirements/base.txt]: why was this commented out in the first place? Does anyone know?

File: seed/energy/meter_data_processor/green_button_data_analyser.py [hunk 3]
```diff
@@ -2,8 +2,8 @@
 from datetime import date, datetime
 from time import sleep
 
+import tasks as aggregator
```
Comment (nllong) [inline seed/energy/meter_data_processor/green_button_data_analyser.py]: I am not a fan of renaming the imports unless it is to shorten it.

File: seed/views/main.py [hunk 4]
```diff
@@ -2376,16 +2377,18 @@ def save_gb_request_info(request):
...
         last_date = yesterday.strftime(date_pattern)
         if loopback_flag == 'Y':
-            very_first_date = datetime.strptime('1/1/2014', '%m/%d/%Y').date()
+            very_first_date = datetime.strptime('3/1/2014', '%m/%d/%Y').date()
             last_date = very_first_date.strftime(date_pattern)
     else:
         last_date = str(calendar.timegm(time.strptime(yesterday.strftime('%m/%d/%Y'), '%m/%d/%Y')))
         if loopback_flag == 'Y':
-            last_date = str(1388534400)
+            last_date = str(1393632000)
```
Comment (nllong) [inline seed/views/main.py]: can you add a comment to what this "last_date" is?

File: seed/views/main.py [hunk 5]
```diff
@@ -2376,16 +2377,18 @@ def save_gb_request_info(request):
     if time_type == 'date':
         last_date = yesterday.strftime(date_pattern)
         if loopback_flag == 'Y':
-            very_first_date = datetime.strptime('1/1/2014', '%m/%d/%Y').date()
+            very_first_date = datetime.strptime('3/1/2014', '%m/%d/%Y').date()
```
Comment (nllong) [inline seed/views/main.py]: what if we want to import files older than 3/1/2014?

Comment (wanghp18) [inline seed/views/main.py]: I will remove the hard code start date later, currently the GreenButton
service I'm using sometimes returns '204 no content' if the start date is
set too early. I will try to load the backfill data month by month
backwardly until there is no data anymore.

File: seed/views/main.py (hunk 4 above)
Comment (wanghp18) [inline seed/views/main.py]: I will add the comments.

File: config/settings/common.py (hunk 1 above)
Comment (wanghp18) [inline config/settings/common.py]: I move the tasks.py to avoid circule import situation, tasks.py needs to
call data analyze function in
energy/meter_data_processor/green_button_data_analyser,
and energy/meter_data_processor/green_button_data_analyser needs to call
the monthly data aggregate function in tasks.py.

File: requirements/base.txt (hunk 2 above)
Comment (wanghp18) [inline requirements/base.txt]: I think django-celery has been commented out quite a while...

Comment (nllong) [inline requirements/base.txt]: strange--okay. just wondering.

File: seed/views/main.py (hunk 5 above)
Comment (nllong) [inline seed/views/main.py]: ok. might want to add a comment.

File: config/settings/common.py (hunk 1 above)
Comment (nllong) [inline config/settings/common.py]: i don't think there is a circular issue. leave it for now. I may move it later.

File: requirements/base.txt (hunk 2 above)
Comment (nicholasserra) [inline requirements/base.txt]: We're running the celery package straight now, no need for django-celery, this shouldn't be uncommented.

File: config/settings/common.py [hunk 6]
```diff
@@ -87,6 +87,8 @@
     'raven.contrib.django.raven_compat',
     'tos',
     'rest_framework',
+    
+    'django',
```
Comment (nicholasserra) [inline config/settings/common.py]: Why was this added? Looks incorrect.

File: requirements/base.txt (hunk 2 above)
Comment (wanghp18) [inline requirements/base.txt]: I followed docs/source/LINUX.setup.rst to deploy SEED, if django-celery is
missing, command 'python manage.py celery worker -B -c 2 --loglevel=INFO -E
--maxtasksperchild=1000' will return 'command celery is not found' error.
Do we have a new command to run Celery?

Comment (nicholasserra) [inline requirements/base.txt]: Yes, apologies for old docs

`DJANGO_SETTINGS_MODULE=config.settings.dev celery -A seed worker -l info -c 4 --maxtasksperchild=1000 --events`

File: config/settings/common.py (hunk 6 above)
Comment (wanghp18) [inline config/settings/common.py]: Sorry, it is supposed to be 'djcelery', if django-celery is not needed, it
can be deleted.

File: requirements/base.txt (hunk 2 above)
Comment (wanghp18) [inline requirements/base.txt]: Thanks, I will delete django-celery.
